gen/service_test.go

```diff
@@ -0,0 +1,402 @@
+			},
+		},
+		{
+			desc: "getValue failure",
+			run: func() (interface{}, error) {
+				return keyvalue.GetValue.WrapResponse(nil, errors.New("foo"))
+			},
+			expectedError: errors.New("foo"),
+		},
+		{
+			desc: "deleteValue success",
+			run: func() (interface{}, error) {
+				return keyvalue.DeleteValue.WrapResponse(nil)
+			},
+			expectedResult: &keyvalue.DeleteValueResult{},
+		},
+		{
+			desc: "deleteValue application error (1)",
+			run: func() (interface{}, error) {
+				return keyvalue.DeleteValue.WrapResponse(&tx.DoesNotExistException{Key: "foo"})
+			},
+			expectedResult: &keyvalue.DeleteValueResult{
+				DoesNotExist: &tx.DoesNotExistException{Key: "foo"},
+			},
+		},
+		{
+			desc: "deleteValue application error (2)",
+			run: func() (interface{}, error) {
+				return keyvalue.DeleteValue.WrapResponse(&tv.InternalError{})
+			},
+			expectedResult: &keyvalue.DeleteValueResult{
+				InternalError: &tv.InternalError{},
+			},
+		},
+		{
+			desc: "deleteValue failure",
+			run: func() (interface{}, error) {
+				return keyvalue.DeleteValue.WrapResponse(errors.New("foo"))
+			},
+			expectedError: errors.New("foo"),
+		},
+	}
+
+	for _, tt := range tests {
+		result, err := tt.run()
+		if tt.expectedError != nil {
+			assert.Equal(t, tt.expectedError, err, tt.desc)
+		} else {
+			assert.Equal(t, tt.expectedResult, result, tt.desc)
+		}
+	}
+}
+
+func TestUnwrapResponse(t *testing.T) {
+	tests := []struct {
+		desc           string
+		unwrapResponse interface{}
+		resultArg      interface{}
+
+		expectedReturn interface{}
+		expectedError  error
+	}{
+		{
+			desc:           "setValue success",
+			unwrapResponse: keyvalue.SetValue.UnwrapResponse,
+			resultArg:      &keyvalue.SetValueResult{},
+		},
+		{
+			desc:           "getValue success",
+			unwrapResponse: keyvalue.GetValue.UnwrapResponse,
+			resultArg: &keyvalue.GetValueResult{
+				Success: &tu.ArbitraryValue{BoolValue: boolp(true)},
+			},
+			expectedReturn: &tu.ArbitraryValue{BoolValue: boolp(true)},
+		},
+		{
+			desc:           "getValue failure",
+			unwrapResponse: keyvalue.GetValue.UnwrapResponse,
+			resultArg: &keyvalue.GetValueResult{
+				DoesNotExist: &tx.DoesNotExistException{Key: "foo"},
+			},
+			expectedError: &tx.DoesNotExistException{Key: "foo"},
+		},
+		{
+			desc:           "deleteValue success",
+			unwrapResponse: keyvalue.DeleteValue.UnwrapResponse,
+			resultArg:      &keyvalue.DeleteValueResult{},
+		},
+		{
+			desc:           "deleteValue failure (1)",
+			unwrapResponse: keyvalue.DeleteValue.UnwrapResponse,
+			resultArg: &keyvalue.DeleteValueResult{
+				DoesNotExist: &tx.DoesNotExistException{Key: "foo"},
+			},
+			expectedError: &tx.DoesNotExistException{Key: "foo"},
+		},
+		{
+			desc:           "deleteValue failure (2)",
+			unwrapResponse: keyvalue.DeleteValue.UnwrapResponse,
+			resultArg: &keyvalue.DeleteValueResult{
+				InternalError: &tv.InternalError{},
+			},
+			expectedError: &tv.InternalError{},
+		},
+	}
```

Comment:
:+1: 

Not sure what you mean by the third one, though. You can't have success set for a void method because that field doesn't exist.